config/config.go

```diff
@@ -676,40 +662,94 @@ func (c *SyslogConf) GetFacility() (syslog.Priority, error) {
+	User                   string            `toml:"user,omitempty"`
+	Host                   string            `toml:"host,omitempty"`
+	Port                   string            `toml:"port,omitempty"`
+	KeyPath                string            `toml:"keyPath,omitempty"`
+	KeyPassword            string            `json:"-" toml:"-"`
+	CpeNames               []string          `toml:"cpeNames,omitempty" json:",omitempty"`
+	ScanMode               []string          `toml:"scanMode,omitempty" json:",omitempty"`
+	DependencyCheckXMLPath string            `toml:"dependencyCheckXMLPath,omitempty"`
+	IgnoreCves             []string          `toml:"ignoreCves,omitempty" json:",omitempty"`
+	Containers             *Containers       `toml:"containers,omitempty" json:",omitempty"`
+	UUIDs                  map[string]string `toml:"uuids,omitempty" json:",omitempty"`
+	Memo                   string            `toml:"memo,omitempty"`
 
-	CpeNames               []string
-	DependencyCheckXMLPath string
-
-	// Container Names or IDs
-	Containers Containers
-
-	IgnoreCves []string
+	// For CentOS, RHEL, Amazon
+	Enablerepo []string `toml:",omitempty" json:",omitempty"`
 
 	// Optional key-value set that will be outputted to JSON
-	Optional [][]interface{}
-
-	// For CentOS, RHEL, Amazon
-	Enablerepo []string
+	Optional map[string]interface{} `toml:",omitempty" json:",omitempty"`
 
 	// "pseudo" or ""
 	Type string
 
 	// used internal
-	LogMsgAnsiColor string // DebugLog Color
-	Container       Container
-	Distro          Distro
+	LogMsgAnsiColor string    `toml:"-"` // DebugLog Color
+	Container       Container `toml:"-"`
+	Distro          Distro    `toml:"-"`
+	Mode            ScanMode  `toml:"-"`
 
 	// IP addresses
-	IPv4Addrs []string
-	IPv6Addrs []string
+	IPv4Addrs []string `toml:"-" json:",omitempty"`
+	IPv6Addrs []string `toml:"-" json:",omitempty"`
 }
 
+// ScanMode has a type of scan mode. fast, fast-root, deep and offline
+type ScanMode struct {
+	flag byte
+}
+
+// Set mode
+func (s *ScanMode) Set(f byte) {
+	s.flag |= f
+}
+
+// IsFast return whether scan mode is fast
+func (s ScanMode) IsFast() bool {
+	return s.flag&Fast == Fast
+}
+
+// IsFastRoot return whether scan mode is fastroot
+func (s ScanMode) IsFastRoot() bool {
+	return s.flag&FastRoot == FastRoot
+}
+
+// IsDeep return whether scan mode is deep
+func (s ScanMode) IsDeep() bool {
+	return s.flag&Deep == Deep
+}
+
+// IsOffline return whether scan mode is offline
+func (s ScanMode) IsOffline() bool {
+	return s.flag&Offline == Offline
+}
+
+func (s ScanMode) validate() error {
+	numTrue := 0
+	for _, b := range []bool{s.IsFast(), s.IsFastRoot(), s.IsDeep()} {
+		if b {
+			numTrue++
+		}
+	}
+	if numTrue == 0 {
+		s.Set(Fast)
+	} else if s.IsDeep() && s.IsOffline() {
+		return fmt.Errorf("Don't specify both of -deep and offline")
+	} else if numTrue != 1 {
+		return fmt.Errorf("Specify only one of -fast, -fast-root or -deep")
+	}
+	return nil
+}
+
+const (
+	Fast = byte(1 << iota)
```

Comment:
__[Golint]__

exported const Fast should have comment (or a comment on this block) or be unexported


<!-- Comment by [SideCI](https://sideci.com) -->